### PR TITLE
Support k8s-style secret values in config

### PR DESCRIPTION
**Pull request by sfackler**
## Before this PR
Secret values were always stored inline in configuration files. This doesn't mesh well with deployment in Kuberetes where we want to separate non-sensitive config in ConfigMaps and sensitive config in Secrets.

## After this PR

==COMMIT_MSG==
Added support for referencing secrets in separate files in config.
==COMMIT_MSG==

The server now supports a `${file:...}` syntax to transparently load string-valued fields from external files. To ensure that we're still able to update runtime config when it changes, we track which files are loaded during the deserialization process and compare hashes before deserializing the updated value.

### CODE REVIEW

File: witchcraft-server/src/service/test_util.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ where
 }
 
 thread_local! {
-    static SPANS: RefCell<Vec<Span>> = RefCell::new(vec![]);
+    static SPANS: RefCell<Vec<Span>> = const { RefCell::new(vec![]) };
```
Comment (sfackler) [inline witchcraft-server/src/service/test_util.rs]: Drive-by Clippy fix.

File: witchcraft-server/src/configs.rs [hunk 2]
```diff
@@ -46,15 +51,60 @@ where
...
+                    if *hash != Sha256::digest(&bytes) {
+                        return false;
+                    }
+                }
+                Err(_) => return false,
+            }
+        }
+
+        for path in &self.err_files {
+            if fs::read(path).is_ok() {
```
Comment (sfackler) [inline witchcraft-server/src/configs.rs:87]: We could alternatively say that we're never up to date when we have failed file references.

Comment (TheKeveloper) [reply witchcraft-server/src/configs.rs:87]: I think it's probably preferable to avoid the downstream churn if we just have some file that we know is not valid, so this seems right

**Review by TheKeveloper — APPROVED** [review]
